src/crimson/os/seastore/segment_cleaner.cc

```diff
@@ -304,22 +336,61 @@ SegmentCleaner::rewrite_dirty_ret SegmentCleaner::rewrite_dirty(
   Transaction &t,
   journal_seq_t limit)
 {
-  LOG_PREFIX(SegmentCleaner::rewrite_dirty);
   return ecb->get_next_dirty_extents(
     t,
     limit,
-    config.journal_rewrite_per_cycle
+    config.journal_rewrite_dirty_per_cycle
   ).si_then([=, &t](auto dirty_list) {
+    LOG_PREFIX(SegmentCleaner::rewrite_dirty);
     DEBUGT("rewrite {} dirty extents", t, dirty_list.size());
     return seastar::do_with(
       std::move(dirty_list),
-      [FNAME, this, &t](auto &dirty_list) {
-	return trans_intr::do_for_each(
-	  dirty_list,
-	  [FNAME, this, &t](auto &e) {
+      [this, &t, limit](auto &dirty_list) {
+	journal_seq_t batch_insert_to = dirty_list.empty()
+	  ? limit
+	  : dirty_list.back()->get_dirty_from();
+	return backref_manager.batch_insert_from_cache(
```

Comment:
I think it is better if we can make those two transactions run without one waiting for the other to finish. Otherwise, one current journal-trimming transaction would be split into two sequentially running transactions, which means an extra journal commit has to be waited before a single run the journal trimming to finish and I'm not quite sure whether it'll be better.

Since the current backref implementation is based on the assumption that no concurrent updates can happen, to make the two transaction able to run without one waiting for the other, we need to start the second transaction after the first transaction is submitted and before it's committed, which probably needs more support from the interruptible future machinery than we currently have.

So, I think we can evaluate this propose later, and if it is worthwhile, we can implement it in follow-up PRs. What do you think?